### PR TITLE
xtest: regression: add case 1040

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Adds regression case 1040 to test panic in concurrent open/invoke/close session.

### CODE REVIEW

File: host/xtest/regression_1000.c [hunk 1]
```diff
@@ -3270,7 +3270,69 @@ static void xtest_tee_test_1039(ADBG_Case_t *c)
...
+static void *test_1040_thread(void *arg)
+{
+	TEEC_Operation op = TEEC_OPERATION_INITIALIZER;
+	struct test_1040_thread_arg *a = arg;
+	TEEC_Result res = TEEC_SUCCESS;
+	uint32_t err_orig = 0;
+	TEEC_Session session = { };
+	size_t n = 0;
+
+	while (n < 100) {
```
Comment (jenswi-linaro) [inline host/xtest/regression_1000.c:3290]: We may want to tune this value.

**Comment by jenswi-linaro** [comment]
This test case will fail without https://github.com/OP-TEE/optee_os/pull/6281

File: host/xtest/regression_1000.c [hunk 2]
```diff
@@ -3270,7 +3270,69 @@ static void xtest_tee_test_1039(ADBG_Case_t *c)
...
+		res = TEEC_InvokeCommand(&session, TA_SIMS_CMD_PANIC, &op,
+					 &err_orig);
+		TEEC_CloseSession(&session);
+		if (res != TEEC_ERROR_TARGET_DEAD) {
+			a->res = res;
+			return NULL;
+		}
+		n++;
+	}
+	return NULL;
```
Comment (etienne-lms) [inline host/xtest/regression_1000.c:3313]: explicitly set `a->res = TEE_SUCCESS` to clarifies the test at line 3331?

File: host/xtest/regression_1000.c [hunk 3]
```diff
@@ -3270,7 +3270,69 @@ static void xtest_tee_test_1039(ADBG_Case_t *c)
...
+{
+	struct test_1040_thread_arg arg[NUM_THREADS] = { };
+	size_t nt = NUM_THREADS;
+	size_t n = 0;
+
+	Do_ADBG_BeginSubCase(c, "Concurent invoke with panic in TA");
+	for (n = 0; n < nt; n++) {
+		if (!ADBG_EXPECT(c, 0, pthread_create(&arg[n].thr, NULL,
+						      test_1040_thread,
+						       arg + n)))
```
Comment (etienne-lms) [inline host/xtest/regression_1000.c:3326]: indentation issue

File: host/xtest/regression_1000.c (hunk 2 above)
Comment (jenswi-linaro) [reply host/xtest/regression_1000.c:3317]: OK

**Comment by jenswi-linaro** [comment]
Update

File: host/xtest/regression_1000.c [hunk 4]
```diff
@@ -3270,7 +3270,72 @@ static void xtest_tee_test_1039(ADBG_Case_t *c)
...
+						      test_1040_thread,
+						      arg + n)))
+			nt = n; /* break loop and start cleanup */
+	}
+	for (n = 0; n < nt; n++) {
+		ADBG_EXPECT(c, 0, pthread_join(arg[n].thr, NULL));
+		ADBG_EXPECT_TEEC_SUCCESS(c, arg[n].res);
+	}
+	Do_ADBG_EndSubCase(c, "Concurent invoke with panic in TA");
+
```
Comment (etienne-lms) [inline host/xtest/regression_1000.c:3338]: remove the empty line

**Comment by etienne-lms** [comment]
Could you retrigger CI tests? https://github.com/OP-TEE/optee_os/pull/6281 has been merged.

**Comment by jenswi-linaro** [comment]
Update

**Comment by jenswi-linaro** [comment]
Squashed, rebased and tag applied.

**Pull request merged**